### PR TITLE
SiStripDetInfoFileReader refactoring and cleanup

### PR DESCRIPTION
**Pull request by pieterdavid**
#### PR description:

Following the changes in https://github.com/cms-sw/cmssw/pull/33888 I put `SiStripDetInfoFileReader::kDefaultfile` everywhere we had the hard-coded path, and replaced `SiStripDetInfoFileReader` by a method that reads a file and constructs a `SiStripDetInfo` instance, which simplifies the code in a few places.
I added this reader method as `SiStripDetInfoFileReader::read`, but maybe `kDefaultFile` and `read` could be made static members of `SiStripDetInfo` to simplify things further? Would that be compatible with naming and code rules?

#### PR validation:

Code compiles, matrix tests run without errors.

CC: @robervalwalsh @mdelcourt @mmusich @tsusa

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild please test

File: DQM/TrackerRemapper/test/test_catch2_SiStripTkMaps.cc [hunk 1]
```diff
@@ -13,10 +13,9 @@ TEST_CASE("SiStripTkMaps testing", "[SiStripTkMaps]") {
     gStyle->SetOptStat(0);
     SiStripTkMaps theMap("COLZA L");
     theMap.bookMap("testing SiStripTkMaps", "counts");
-    SiStripDetInfoFileReader reader_ = SiStripDetInfoFileReader(edm::FileInPath(k_geo).fullPath());
-    const std::map<uint32_t, SiStripDetInfoFileReader::DetInfo>& DetInfos = reader_.getAllData();
+    const auto detInfo = SiStripDetInfoFileReader::read(edm::FileInPath(k_geo).fullPath());
```
Comment (mmusich) [inline DQM/TrackerRemapper/test/test_catch2_SiStripTkMaps.cc:16]: can't it be `edm::FileInPath(SiStripDetInfoFileReader::kDefaultFile)` also here?

Comment (pieterdavid) [reply DQM/TrackerRemapper/test/test_catch2_SiStripTkMaps.cc]: Yes, thanks, I didn't notice the definition above

**Comment by mmusich** [comment]
@cmsbuild, please test

File: DQM/TrackerRemapper/test/test_catch2_SiStripTkMaps.cc (hunk 1 above)
Comment (mmusich) [reply DQM/TrackerRemapper/test/test_catch2_SiStripTkMaps.cc]: thanks!

**Comment by jfernan2** [comment]
+1

**Comment by jpata** [comment]
+reconstruction
- only one file in reco (`ClusterizerUnitTesterESProducer.cc`), looks technical
- no reco changes expected or observed

**Comment by ggovi** [comment]
+1

**Comment by yuanchao** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**